keras_hub/src/models/phi4/phi4_decoder.py

```diff
@@ -0,0 +1,246 @@
+import keras
+from keras import ops
+
+from keras_hub.src.layers.modeling.transformer_layer_utils import (
+    compute_causal_mask,
+)
+from keras_hub.src.layers.modeling.transformer_layer_utils import (
+    merge_padding_and_attention_mask,
+)
+from keras_hub.src.models.phi4.phi4_attention import Phi4Attention
+from keras_hub.src.models.phi4.phi4_layernorm import Phi4LayerNorm
+from keras_hub.src.utils.keras_utils import clone_initializer
+
+
+class Phi4Decoder(keras.layers.Layer):
+    """A Transformer decoder layer for the Phi-4 backbone."""
+
+    def __init__(
+        self,
+        hidden_dim=5120,
+        intermediate_dim=17_920,
+        num_query_heads=40,
+        num_key_value_heads=10,
+        activation="silu",
+        layer_norm_epsilon=1e-5,
+        kernel_initializer="glorot_uniform",
+        dropout=0,
+        max_sequence_length=16_384,
+        pretraining_sequence_length=16_384,
+        rope_max_wavelength=250_000,
```

Comment:
Are these default values are mostly common for all the presets in phi-4, if not may be we can remove default values?